.github/workflows/ci.yml

```diff
@@ -8,52 +8,65 @@ on:
 
 defaults:
   run:
-    shell: bash # https://github.com/beeware/briefcase/pull/912
+    shell: bash  # https://github.com/beeware/briefcase/pull/912
 
 jobs:
   verify-apps:
     name: Build apps
+    runs-on: windows-latest
+    timeout-minutes: 20
     strategy:
       fail-fast: false
       matrix:
-        python_version: ["3.8", "3.9", "3.10", "3.11"]
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
+        python-source: ["github", "winstore"]
         framework: ["toga", "pyside2", "pyside6", "ppb"]
-        include:
-          - platform: windows-latest
-            briefcase-data-dir: ~\AppData\Local\BeeWare\briefcase\Cache
-            pip-cache-dir: ~\AppData\Local\pip\Cache
-            docker-cache-dir: C:\ProgramData\DockerDesktop
-
         exclude:
           # PySide2 doesn't publish a binary wheel that is compatible
           # with Python 3.11, and is unlikely to ever do so.
-          - python_version: "3.11"
+          - python-version: "3.11"
             framework: "pyside2"
-
-    runs-on: ${{ matrix.platform }}
     steps:
+
       - name: Cache Briefcase tools
         uses: actions/cache@v3.0.11
         with:
-          key: briefcase-${{ matrix.platform }}-${{ matrix.python_version }}
+          key: briefcase-${{ runner.os }}-${{ matrix.python-version }}
           path: |
             ~/.cookiecutters
-            ${{ matrix.briefcase-data-dir }}
-            ${{ matrix.pip-cache-dir }}
-            ${{ matrix.docker-cache-dir }}
+            ~/AppData/Local/BeeWare/briefcase/Cache
+            ~/AppData/Local/pip/Cache
+
```

Comment:
While caching the Briefcase cache directory nullifies a lot of the rationale for testing against Windows Store Python, it still seemed like a good idea to keep in place.

Github caches are evicted after 7 days. Given this repo doesn't get many PRs, it is likely the action will run without a cache available. I've also included the python source as part of the cache key; so, a cache will only be available for subsequent runs if a previous run succeeded. This helps avoid papering over a failure because the next run would pull the successfully created cache from the github python job.